LanguageFeatures/Extension-types/static_analysis_extension_types_A08_t12.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a type T is incompatible with await if at least one
+/// of the following criteria holds:
+/// - T is an extension type that does not implement Future.
+/// - T is S?, and S is incompatible with await.
+/// - T is X & B, and B is incompatible with await.
+/// - T is a type variable with bound S, and S is incompatible with await.
+/// Consider an expression of the form await e. A compile-time error occurs if
+/// the static type of e is incompatible with await.
+///
+/// @description Checks that it is not an error if `await e` occurs, and the
+/// static type of `e` is `X & B` and `B` does not derive a future type but
```

Comment:
I think this might not be possible: If `X` is incompatible with await then `X` is probably bounded by an extension type that doesn't implement `Future`, and then it probably isn't possible for `B` to be a type that (1) does not derive a future type, and still (2) is compatible with await. In particular, I think `B` _must_ be an extension type (unless it's a bottom type, but if we promote to a bottom type then the code is unreachable), and an extension type is not compatible with await unless it implements `Future`, and in that case it also derives a future type.